aineko/core/dataset.py

```diff
@@ -132,30 +140,120 @@ def _validate_message(
+        """Polls a message from the dataset.
 
         Args:
-            how: how to read the message
+            how: how to read the message.
                 "next": read the next message in the queue
+                "last": read the last message in the queue
+            timeout: seconds to poll for a response from kafka broker.
+                If using how="last", set to bigger than 0.
 
         Returns:
-            msg: message from the dataset
+            message from the dataset
+
+        Raises:
+            ValueError: if how is not "next" or "last"
         """
+        if how not in ["next", "last"]:
+            raise ValueError(f"Invalid how: {how}. Expected `next` or `last`.")
+
         timeout = timeout or self.kafka_config.get("CONSUMER_TIMEOUT")
         if how == "next":
             # next unread message from queue
-            message = self._validate_message(
-                self.consumer.poll(timeout=timeout)
-            )
-        else:
-            raise ValueError(f"Invalid how: {how}. Expected 'next'.")
+            message = self.consumer.poll(timeout=timeout)
+
+        if how == "last":
+            # last message from queue
+            self._update_offset_to_latest()
+            message = self.consumer.poll(timeout=timeout)
+
+        self.cached = True
+
+        return self._validate_message(message)
+
+    def _consume_message(
+        self, how: Literal["next", "last"], timeout: Optional[float] = None
+    ) -> dict:
+        """Calls the consume method and blocks until a message is returned.
+
+        Args:
+            how: See `consume` method for available options.
+
+        Returns:
+            message from dataset
+        """
+        while True:
+            try:
+                message = self.consume(how=how, timeout=timeout)
+                if message is not None:
+                    return message
+            except KafkaError as e:
+                if e.code() == "_MAX_POLL_EXCEEDED":
+                    continue
+                raise e
+
+    def next(self) -> dict:
+        """Use to consume the next message from the dataset.
+
+        This method is wraps `consume(how="next")`. It implements a
+        block that wait until a message is recieved before returning it.
+        This is useful when the timeout is short and the consumer often
+        returns `None`.
 
-        return message
+        Returns:
+            msg: message from the dataset
+        """
+        return self._consume_message(how="next")
+
+    def last(self, timeout: int = 1) -> dict:
+        """Used to consume the last message from the dataset.
+
+        This method is wraps `consume(how="next")`. It implements a
+        block that wait until a message is recieved before returning it.
+        This is useful when the timeout is short and the consumer often
+        returns `None`. The timeout must be >0 to prevent overwhelming the
+        broker with requests to update the offset.
```

Comment:
```suggestion
        This method wraps the `consume(how="next")` function. It implements a
        block that waits until a message is received before returning it.
        This is useful when the timeout is short and the consumer often
        returns `None`. The timeout must be greater than 0 to prevent
        overwhelming the broker with requests to update the offset.
```
Slight edit + typo fix.

Reply:
I don't understand the difference between next and last, based on the docstrings.